python/rapidsmpf/rapidsmpf/buffer/packed_data.pyx

```diff
@@ -13,3 +15,39 @@ cdef class PackedData:
         cdef PackedData self = PackedData.__new__(PackedData)
         self.c_obj = move(obj)
         return self
+
+    @staticmethod
```

Comment:
Make it a `@classmethod` and do `cls.__new__(cls)` like in the spill manager: https://github.com/rapidsai/rapidsmpf/blob/b77ed56a56a357d4e1f1bcb6208098a8078fb740/python/rapidsmpf/rapidsmpf/buffer/spill_manager.pyx#L95C33